crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -102,12 +91,74 @@ impl<Wg: WireGuard, E: Backoff> StunEndpointProvider<Wg, E> {
                 ping_pong_tracker: ping_pong_handler,
                 stun_peer_publisher,
                 stun_state: StunState::WaitingForWg,
+                sockets: None,
             }),
         }
     }
 
-    pub async fn configure(&self, mut servers: Vec<Server>) {
+    pub async fn configure(
+        &self,
+        servers: Vec<Server>,
+        use_ipv6: bool,
+        socket_pool: Arc<SocketPool>,
+    ) {
+        self.configure_with_ext_socket_addr(servers, use_ipv6, socket_pool, Ipv4Addr::UNSPECIFIED)
+            .await
+    }
+
+    async fn configure_with_ext_socket_addr(
+        &self,
+        mut servers: Vec<Server>,
+        use_ipv6: bool,
+        socket_pool: Arc<SocketPool>,
+        ext_socket_addr: Ipv4Addr,
+    ) {
         let _ = task_exec!(&self.task, async move |s| {
+            let ip_version_changed = s
+                .sockets
+                .as_ref()
+                .and_then(|sockets| {
+                    sockets
+                        .tun_socket
+                        .local_addr()
+                        .map(|addr| addr.is_ipv6() != use_ipv6)
+                        .ok()
+                })
+                .unwrap_or(false);
+
+            if s.sockets.is_none() || ip_version_changed {
+                s.sockets = match (
+                    if use_ipv6 {
```

Comment:
NIT: i would like to see these moved to var's above match, It would personaly for me look less busy

Reply:
Updated